packages/sync/src/replicastore/class-table-checksum.php

```diff
@@ -0,0 +1,386 @@
+<?php
+
+namespace Automattic\Jetpack\Sync\Replicastore;
+
+use Automattic\Jetpack\Sync\Settings;
+use Exception;
+use WP_error;
+
+// TODO add rest endpoints to work with this, hopefully in the same folder
+
+class Table_Checksum {
+	public $table                 = '';
+	public $table_configuration   = array();
+	public $range_field           = '';
+	public $key_fields            = array();
+	public $checksum_fields       = array();
+	public $additional_filter_sql = '';
+
+
+	public $default_tables = array();
```

Comment:
is public appropriate for this and allowed tables? should we have private/protected also what about others should they be accessible?

Reply:
I prefer keeping these properties public to allow for easier unit testing (changing the input values). 

Moving to private/protected adds a layer of complexity when doing some simpler tests.